tests/suites/test_suite_ssl.function

```diff
@@ -339,6 +341,230 @@ int mbedtls_mock_tcp_recv_nb( void *ctx, unsigned char *buf, size_t len )
     return mbedtls_test_buffer_get( socket->input, buf, len );
 }
 
+
+/*
+ * Structure with endpoint certificates for SSL comunication tests.
+ */
+typedef struct mbedtls_endpoint_certificate{
+    mbedtls_x509_crt ca_cert;
+    mbedtls_x509_crt cert;
+    mbedtls_x509_crt cert2;
+    mbedtls_pk_context pkey;
+    mbedtls_pk_context pkey2;
+} mbedtls_endpoint_certificate;
+
+/*
+ * Endpoint structure for SSL comunication tests.
+ */
+typedef struct mbedtls_endpoint{
+        const char *name;
+        mbedtls_ssl_context ssl;
+        mbedtls_ssl_config conf;
+        mbedtls_ctr_drbg_context ctr_drbg;
+        mbedtls_entropy_context entropy;
+        mbedtls_mock_socket socket;
+        mbedtls_endpoint_certificate *cert;
+} mbedtls_endpoint;
+
+#define EP( VAR )   ( &( ep -> VAR ) )
+#define CERT( VAR ) ( &( ep_cert -> VAR ) )
+
+/*
+ * Initialises \p ep structure.
+ * \p endpoint_type mast be set as MBEDTLS_SSL_IS_SERVER or
+ * MBEDTLS_SSL_IS_CLIENT.
+ *
+ * \retval  0 on success or other value on error
+ */
+int mbedtls_endpoint_init( mbedtls_endpoint *ep, int endpoint_type )
+{
+    int ret = 0;
+
+    ep->name = ( endpoint_type == MBEDTLS_SSL_IS_SERVER ) ? "Server" : "Client";
+
+    mbedtls_ssl_init( EP(ssl) );
+    mbedtls_ssl_config_init( EP(conf) );
+    mbedtls_ctr_drbg_init( EP(ctr_drbg) );
+    mbedtls_ssl_conf_rng( EP(conf),
+        mbedtls_ctr_drbg_random,
+        EP(ctr_drbg) );
+    mbedtls_entropy_init( EP(entropy) );
+    mbedtls_mock_socket_init( EP(socket));
+
+    if( ( ret = mbedtls_ctr_drbg_seed( EP( ctr_drbg ), mbedtls_entropy_func,
+                    EP( entropy ), ( const unsigned char* ) ( ep->name ),
+                    strlen( ep->name ) ) ) != 0 )
+    {
+        return ret;
+    }
+
+    mbedtls_ssl_set_bio( EP(ssl), EP(socket),
+        mbedtls_mock_tcp_send_b,
+        mbedtls_mock_tcp_recv_b,
+        NULL );
+
+    if( ( ret = mbedtls_ssl_setup( EP(ssl), EP(conf) ) ) != 0 )
+    {
+        return ret;
```

Comment:
Brackets can be omitted for simple one-line conditions with one line of effect following them.

Reply:
Yes, I know. But because sometimes they can be omitted and sometimes, they cannot (multi-line conditions) I prefer to add them always. I think it looks a little bit oddly for someone how does not know our coding standards that sometimes we use brackets and sometimes we do not. Besides, it is better to debug the code with brackets